packages/compass-connections/src/index.spec.tsx

```diff
@@ -0,0 +1,346 @@
+      expect(
+        screen.getByTestId('connection-error-summary').textContent
+      ).to.contain(expectedErrorText);
+    };
+
+    render(
+      <ToastArea>
+        <Connections
+          connectionStorage={
+            {
+              loadAll: () => Promise.resolve([]),
+              save: () => Promise.resolve(),
+              delete: () => Promise.resolve(),
+              load: () => Promise.resolve(undefined),
+            } as any
+          }
+          appName="Test App"
+          connectFn={connectSpy}
+          onConnected={() => {}}
+        />
+      </ToastArea>
+    );
+
+    expect(connectSpy).not.to.have.been.called;
+  });
+
+  afterEach(cleanup);
+
+  it('connects to localhost by default', async function () {
+    await expectToConnectWith({
+      connectionString: 'mongodb://localhost:27017/?appName=Test+App',
+    });
+  });
+
+  describe('In-Use Encryption', function () {
+    beforeEach(async function () {
+      if (process?.env?.COMPASS_CSFLE_SUPPORT !== 'true') {
+        this.skip();
+      }
+
+      await openAdvancedTab('csfle');
+      setInputValue('csfle-keyvault', 'db.coll');
+    });
+
+    it('passes the FLE options if csfle-keyvault is set', async function () {
+      await expectToConnectWith({
+        connectionString: 'mongodb://localhost:27017/?appName=Test+App',
+        fleOptions: {
+          storeCredentials: false,
+          autoEncryption: { keyVaultNamespace: 'db.coll' },
+        },
+      });
+    });
+
+    it('sets storeCredentials', async function () {
+      fireEvent.click(screen.getByTestId('csfle-storeCredentials-input'));
+
+      await expectToConnectWith({
+        connectionString: 'mongodb://localhost:27017/?appName=Test+App',
+        fleOptions: {
+          storeCredentials: true,
+          autoEncryption: { keyVaultNamespace: 'db.coll' },
+        },
+      });
+    });
+
+    it('reports an error if the key vault namespace is not set', async function () {
+      setInputValue('csfle-keyvault', '');
+
+      setEditorValue(
+        screen.getByTestId('encrypted-fields-map-editor'),
+        '{ "db.coll": { fields: [] } }'
+      );
+
+      await expectConnectionError(
+        'Key Vault namespace must be specified for In-Use-Encryption-enabled connections'
+      );
+    });
+
+    it('allows to set encrypted fields map', async function () {
+      setEditorValue(
+        screen.getByTestId('encrypted-fields-map-editor'),
+        '{ "db.coll": { fields: [] } }'
+      );
+
+      await expectToConnectWith({
+        connectionString: 'mongodb://localhost:27017/?appName=Test+App',
+        fleOptions: {
+          storeCredentials: false,
+          autoEncryption: {
+            keyVaultNamespace: 'db.coll',
+            encryptedFieldsMap: {
+              '$compass.error': null,
+              '$compass.rawText': '{ "db.coll": { fields: [] } }',
+              'db.coll': { fields: [] },
```

Comment:
I think one of the things that we should check here if we can is that this actually allows including BSON values, not just JSON, e.g. `fields: [{path: 'foo', bsonType: 'string', keyId: UUID('11d58b8a-0c6c-4d69-a0bd-70c6d9befae9') }]`

Reply:
Tested the `UUID`, is there any known reason why `expect(keyid).to.deep.equal(new bson.UUID('11d58b8a-0c6c-4d69-a0bd-70c6d9befae9'))` didn't work while `Binary` did instead?

Reply:
Yeah, `bson.UUID` is actually not a BSON type, it’s more like a generic UUID class. `new bson.UUID(...).toBinary()` should work :slightly_smiling_face: